### PR TITLE
Updated to use 0.4.0 version of IntegrationTesting package

### PR DESCRIPTION
**Pull request by kichalla**

### CODE REVIEW

File: test/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.FunctionalTests/Infrastructure/ApplicationTestFixture.cs [hunk 1]
```diff
@@ -30,24 +31,18 @@ protected ApplicationTestFixture(string applicationName)
 
         public ILogger Logger { get; private set; }
 
-        public DeploymentResult CreateDeployment()
-        {
-            lock (_deploymentLock)
```
Comment (kichalla) [inline test/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.FunctionalTests/Infrastructure/ApplicationTestFixture.cs]: Curious why was this required earlier? All the tests publish to different output folders, so even if they run in parallel it shouldn't be a problem

File: test/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.FunctionalTests/ApplicationWithParseErrorsTest.cs [hunk 2]
```diff
@@ -33,14 +34,14 @@ public void PublishingPrintsParseErrors()
...
 
             // Assert
-            var output = Fixture.TestSink.Writes.Select(w => w.State.ToString().Trim()).ToList();
+            var logs = Fixture.TestSink.Writes.Select(w => w.State.ToString().Trim()).ToList();
 
-            foreach (var error in expectedErrors)
+            foreach (var expectedError in expectedErrors)
             {
-                Assert.Contains(error, output);
+                Assert.True(logs.Any(log => log.Contains(expectedError)));
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.FunctionalTests/ApplicationWithParseErrorsTest.cs]: `Assert.Contains(expectedError, logs)`

File: global.json [hunk 3]
```diff
@@ -0,0 +1 @@
+{ "sdk": { "version": "2.0.0-preview1-005783" } }
```
Comment (pranavkm) [inline global.json]: .gitignore

File: test/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.FunctionalTests/Infrastructure/ApplicationTestFixture.cs (hunk 1 above)
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.FunctionalTests/Infrastructure/ApplicationTestFixture.cs]: Tests in the same fixture use the same output.

File: test/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.FunctionalTests/Infrastructure/ApplicationTestFixture.cs [hunk 4]
```diff
@@ -30,24 +31,18 @@ protected ApplicationTestFixture(string applicationName)
...
-                {
-                    return _deploymentResult;
-                }
+        public ILoggerFactory LoggerFactory { get; private set; }
 
-                Logger = CreateLogger();
-                var deploymentParameters = GetDeploymentParameters();
-                var deployer = ApplicationDeployerFactory.Create(deploymentParameters, Logger);
-                _deploymentResult = deployer.Deploy();
+        public async Task<DeploymentResult> CreateDeploymentAsync()
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.FunctionalTests/Infrastructure/ApplicationTestFixture.cs]: We might need the lock as yet. Do we run the tests in parallel or does the deployer manage concurrency?

File: test/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.FunctionalTests/Infrastructure/ApplicationTestFixture.cs (hunk 1 above)
Comment (kichalla) [inline test/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.FunctionalTests/Infrastructure/ApplicationTestFixture.cs]: Since each test is calling `CreateDeploymentAsync`, they would have a separate output folder for each of them. This is how it should have been before my change.
